File: hw2/RaceMonitor_Java/src/Messages/FinishedUpdate.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Messages;

import Exceptions.ApplicationException;
import Racedata.AthleteRaceStatus;
import java.time.LocalDateTime;
/**
 *
 * @author Shimin
 * @author Stephen Clyde
 */
public class FinishedUpdate extends AthleteUpdate {
    private LocalDateTime officialEndTime;
    
    public FinishedUpdate() {
        super(AthleteRaceStatus.Finished);
    }
    
    public FinishedUpdate(String[] properties) throws ApplicationException {
        super (AthleteRaceStatus.Finished, properties);
        
        if (properties.length != 4 || !properties[0].equals(AthleteRaceStatus.Finished.toString()))
            throw new ApplicationException("Invalid properties");
        
        officialEndTime = LocalDateTime.parse(properties[3], dateTimeFormatter);
    }

    public String toString() { return super.toString() + "," + officialEndTime.format(dateTimeFormatter); }
    
    public void setOfficialEndTime (LocalDateTime OfficialEndTime) {
        officialEndTime = OfficialEndTime;
    }

    public LocalDateTime getOfficialEndTime () {
        return officialEndTime;
    }

}

File: hw2/RaceMonitor_CSharp/RaceData/Messages/FinishedUpdate.cs
using System;

namespace RaceData.Messages
{
    public class FinishedUpdate :AthleteUpdate
    {
        public DateTime OfficialEndTime { get; set; }

        public FinishedUpdate() : base(AthleteRaceStatus.Finished) { }

        public FinishedUpdate(string[] properties) : base(AthleteRaceStatus.Finished, properties)
        {
            if (properties.Length != 4 || properties[0] != AthleteRaceStatus.Finished.ToString())
                throw new ApplicationException("Invalid properties");

            OfficialEndTime = Convert.ToDateTime(properties[3]);
        }

        public override string ToString()
        {
            return $"{base.ToString()},{OfficialEndTime}";
        }
    }

}
